pkg/utils/util.go

```diff
@@ -9,3 +16,70 @@ func Contains(slice []string, value string) bool {
 	}
 	return false
 }
+
+// GetToolchainClusterName produces a name for ToolchainCluster object that is both deterministic and "reasonably unique".
+// The `ordinal` must be greater than 0 if there are multiple ToolchainCluster objects pointing to the same cluster. This
+// needs to be determined by the caller prior to calling this method.
+func GetToolchainClusterName(clusterType, serverAPIEndpoint string, ordinal int) (string, error) {
+	// NOTE: this function is ported from the original add-cluster.sh script to produce the same names during the transition
+	// period to the new operator-based approach to the member registration.
+	// Since add-cluster.sh was a bash script with a long history, the logic is a bit convoluted at places (especially in
+	// handling the numerical suffix).
+
+	// we need to make sure that:
+	// 1) the name is at most 63 characters long
+	// 2) the variable part is (a part of) the cluster hostname
+	// 3) it ends with a digit (supplied by the ordinal param) if it was shortened
+
+	var ordinalLength int
+	if ordinal > 0 {
+		// log10(x) + 1 determines the number of digits in a number (in base 10),
+		// i.e. the length of the suffix that we are potentially adding to the name
+		ordinalLength = int(math.Log10(float64(ordinal))) + 1
+	} else {
+		ordinalLength = 0
+	}
+
+	// the name always contains the cluster type, a hypen between the cluster type and the numerical suffix (if needed)
+	// Interestingly, this is computed BEFORE we determine if we need the numerical suffix at all, but that's the logic
+	// in the original script.
```

Comment:
I wanted to keep the logic as-is as much as possible so that we don't miss some weird corner case. But since we're redoing it quite a bit, I guess I can think about some simplifications. Testsuite is the key :)